validation/src/CueVetDiagnosticsProvider.ts

```diff
@@ -0,0 +1,130 @@
+import * as vscode from 'vscode';
+import { DiagnosticProvider } from './DiagnosticsProvider';
+import { spawn } from 'child_process';
+import { writeFileSync } from 'fs';
+import { mkdtempSync, rmdirSync } from 'node:fs';
+import { join } from 'node:path';
+import { tmpdir } from 'node:os';
+import { randomBytes } from 'crypto';
+
+export class CueVetDiagnosticsProvider implements DiagnosticProvider {
+    private collection: vscode.DiagnosticCollection
+
+    private mockContext = `#Context: {
+        appRevision:    string
+        appRevisionNum: int
+        appName:        string
+        name:           string
+        namespace:      string
+        output:         _
+    }
+    
+    context: #Context & {
+        appRevision:    "v1"
+        appRevisionNum: 1
+        appName:        "dummy-app-name"
+        name:           "dummy-name"
+        namespace:      "dummy-namespace"
+        output:         {}
+    }`;
+
+    private tempDirectory: string | undefined;
+
+    constructor(collection: vscode.DiagnosticCollection) {
+        this.collection = collection;
+    }
+
+    activate(): void {
+        const directory = mkdtempSync(join(tmpdir(), 'vela-vscode-extension-'));
+        console.log(directory);
+        this.tempDirectory = directory;
+    }
+
+    deactivate(): void {
+        if (this.tempDirectory) {
+            rmdirSync(this.tempDirectory);
+        }
+    }
+
+    getName(): string {
+        return 'cue';
+    }
+
+    // This is a temporary measure to supress certain messages until all of the data injected by Kubevela controller can be mocked.
+    // So far only context has been mocked.
+    private shouldTemporarilyIgnore(problem: string): boolean {
+        const regexes = [
+            // some instances are incomplete; use the -c flag to show errors or suppress this message
+            /some instances are incomplete/,
+            // template.output.metadata.name: reference "parameter" not found
+            /reference "parameter" not found/
+        ];
```

Comment:
Would also be keen to hear a better way of suppressing those error messages. This implementation means, that if you template errors out with one of these, no other legitimate errors will be shown.